### PR TITLE
print file content

### PR DESCRIPTION
**Pull request by thebioengineer**
Add internal functions for printing file contents sans roxygen./yaml headers. Should make it simpler for building the vignette in the `vt_use_report()` function being built in #45

### CODE REVIEW

File: tests/testthat/test-add_file.R [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      file = "example_req.md",
+      report = "report_dynamic.Rmd",
+      dynamic_referencing = TRUE)
+    add_file_to_report(
+      file = "example_code.r",
+      report = "report_dynamic.Rmd",
+      dynamic_referencing = TRUE)
+
+
+    report_text <- readLines("report.Rmd")
```
Comment (mariev) [inline tests/testthat/test-add_file.R:199]: knitting this `report.Rmd` document does not preserve markdown formatting

image

**Review by mariev — CHANGES_REQUESTED** [review]
1. test cases break after bringing in `develop` branch
2. when knitting the report.Rmd document, rmarkdown formatting isn't always preserved, i.e. bullets, etc. I think we need a test case that takes us from `example_req.md` to `sample.html` 
3. do we need to be concerned about absolute file paths?

File: tests/testthat/test-add_file.R (hunk 1 above)
Comment (thebioengineer) [reply tests/testthat/test-add_file.R:199]: 🤔 interesting. It should be returning "asis". I'll look into it

**Comment by thebioengineer** [comment]
passes all checks locally, and know prints out nicely. Trick was to set the chunk option to 'asis' and cat as opposed to print.

**Comment by thebioengineer** [comment]
What do you mean by being concerned about absolute file paths? Everything is relative to `vt_path()`

**Comment by mariev** [comment]
file paths structured as `requirements/example_req.md`

**Comment by thebioengineer** [comment]
ah, i see. yeah, we could make `vt_file` more like `add_file` so just the filename is passed as opposed to the location. I was thinking writing it this way mean more flexibility, but now that I think about it, that might not be good

**Comment by mariev** [comment]
I would just update to using `file.path` in the tests cases? I believe the existing functions will work regardless of whether users are following good programming practices

File: tests/testthat/test-add_file.R [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+        "vt_kable_test_code(results)",
+        "```"
+      )
+    )
+
+    expect_equal(
+      report_dynamic_text,
+      c("",
+        "```{r example-req, echo = FALSE, results = 'asis'}",
+        "vt_file(file=vt_path(\"requirements/example_req.md\"), dynamic_referencing = TRUE)",
```
Comment (mariev) [inline tests/testthat/test-add_file.R:221]:
```suggestion
        "vt_file(file=vt_path(file.path(\"requirements\"), \"example_req.md\")), dynamic_referencing = TRUE)",
```

**Comment by thebioengineer** [comment]
Hey Marie, I think your issues have now all been addressed?

**Review by mariev — APPROVED** [review]

**Pull request merged**